pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -376,7 +377,9 @@ func (s *Service) currentState(thid string) (state, error) {
 }
 
 func (s *Service) update(thid string, state state) error {
-	err := s.store.Put(thid, []byte(state.Name()))
+	// TODO replace with the connection id : issue-420
```

Comment:
Question: is it always save new record? or sometimes you update existing connection record?

If always save new record then --> function name is misleading.
If update --> function name is misleading + persistence should update existing record?

Reply:
if it is existing then it may replace already saved record, correct me if I am wrong... I am talking about key-val in store.

Reply:
For example, states get updated with every transition but connection/ thread id for now remains the same per instance.